packages/connect-core/src/utils/descriptor/index.ts

```diff
@@ -0,0 +1,45 @@
+import { AppOrAddress, StepDescribed } from '../../types'
+import App from '../../entities/App'
+
+type ForwardingPathDescriptionTreeEntry =
+  | AppOrAddress
+  | [AppOrAddress, ForwardingPathDescriptionTreeEntry[]]
+
+type ForwardingPathDescriptionTree = ForwardingPathDescriptionTreeEntry[]
```

Comment:
I leave the definition of the types from the specification but I'm not sure if it's still relevant given the new changes.